compiler/modules/nback/structure.bal

```diff
@@ -442,6 +462,16 @@ function mappingFieldIndex(t:Context tc, t:SemType mappingType, bir:StringOperan
     return ();
 }
 
+function openMappingFieldIndex(t:Context tc, t:SemType mappingType, bir:StringOperand k) returns int? {
```

Comment:
Instead of having a separate function, make `mappingFieldIndex` return a record saying whether it's closed and the index.

Reply:
Changed